client/state/selectors/is-connected-secondary-network-site.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+import { isMainSiteOf } from 'state/selectors';
+import { isJetpackSiteSecondaryNetworkSite } from 'state/sites/selectors';
+
+/**
+ * Returns true if site with id equal to siteId is a connected secondary network site and false otherwise
+ * Returns null if with the information available it's not possible to determine if site is a connected secondary site or not
+ * In order for a site to be a connected secondary site its main site also needs to be connected
+ * With selector isJetpackSiteSecondaryNetworkSite the main site does not need to be connected
+ *
+ * @param  {Object}    state     Global state tree
+ * @param  {Number}    siteId    The ID of the site we're querying
+ * @return {?Boolean}            Whether site with id equal to siteId is a connected secondary network site
+ */
+export default createSelector(
+	( state, siteId ) => {
+		const isSecondaryNetworkSite = isJetpackSiteSecondaryNetworkSite( state, siteId );
+		if ( isSecondaryNetworkSite === null ) {
+			return null;
+		}
+		if ( ! isSecondaryNetworkSite ) {
+			return false;
+		}
+		const siteIds = Object.keys( get( state, 'sites.items', {} ) );
+		return siteIds.reduce( ( previousResult, mainSiteId ) => {
+			if ( previousResult ) {
+				return previousResult;
```

Comment:
Which makes me think: why would we want to consider the `null` a truthful value? If we don't it would be even simpler:

```
some( siteIds, ( mainSiteId ) => isMainSiteOf( state, mainSiteId, siteId ) )
```